File: textUtilsLib/src/main/java/com/shuyu/textutillib/span/LinkSpan.java
package com.shuyu.textutillib.span;


import android.content.Context;
import android.text.TextPaint;
import android.text.TextUtils;
import android.text.style.ClickableSpan;
import android.view.View;

import com.shuyu.textutillib.listener.SpanUrlCallBack;

/**
 * 显示可点击的手机号码和URL
 * Created by shuyu on 2016/11/10.
 */
public class LinkSpan extends ClickableSpan {

    private String url;
    private Context context;
    private SpanUrlCallBack spanUrlCallBack;
    private int color;

    public LinkSpan(Context context, String url, int color, SpanUrlCallBack spanUrlCallBack) {
        this.url = url;
        this.context = context;
        this.spanUrlCallBack = spanUrlCallBack;
        this.color = color;
    }

    @Override
    public void onClick(View widget) {
        if ((url.contains("tel:") && TextUtils.isDigitsOnly(url.replace("tel:", ""))) || TextUtils.isDigitsOnly(url)) {
            if (spanUrlCallBack != null)
                spanUrlCallBack.phone(widget, url);
        } else {
            if (spanUrlCallBack != null)
                spanUrlCallBack.url(widget, url);
        }
    }


    @Override
    public void updateDrawState(TextPaint ds) {
        ds.setColor(color);
        /** 去掉下划线 ， 默认自带下划线 **/
        ds.setUnderlineText(false);
    }
}

File: textUtilsLib-kotlin/src/main/java/com/shuyu/textutillib/span/LinkSpan.kt
package com.shuyu.textutillib.span


import android.content.Context
import android.text.TextPaint
import android.text.TextUtils
import android.text.style.ClickableSpan
import android.view.View

import com.shuyu.textutillib.listener.SpanUrlCallBack

/**
 * 显示可点击的手机号码和URL
 * Created by shuyu on 2016/11/10.
 */
open class LinkSpan(private val context: Context, private val url: String, private val color: Int, private val spanUrlCallBack: SpanUrlCallBack?) : ClickableSpan() {

    override fun onClick(widget: View) {
        if (url.contains("tel:") && TextUtils.isDigitsOnly(url.replace("tel:", "")) || TextUtils.isDigitsOnly(url)) {
            spanUrlCallBack?.phone(widget, url)
        } else {
            spanUrlCallBack?.url(widget, url)
        }
    }


    override fun updateDrawState(ds: TextPaint) {
        ds.color = color
        /** 去掉下划线 ， 默认自带下划线  */
        ds.isUnderlineText = false
    }
}